File: backend/src/main/java/com/coigniez/resumebuilder/domain/column/ColumnRequest.java
package com.coigniez.resumebuilder.domain.column;

import java.util.ArrayList;
import java.util.List;

import com.coigniez.resumebuilder.domain.columnsection.ColumnSectionRequest;
import com.coigniez.resumebuilder.domain.layout.enums.ColorLocation;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.*;


@Data
@Builder
public class ColumnRequest {

    private Long id;

    private long layoutId;

    @Min(1) @Max(2)
    private Integer columnNumber;

    @Builder.Default
    @NotNull
    private List<ColumnSectionRequest> sectionMappings = new ArrayList<>();

    private ColorLocation backgroundColor;
    private ColorLocation textColor;
    private ColorLocation borderColor;

    @Min(0)
    @Builder.Default
    private double paddingLeft = 10.0;
    @Min(0)
    @Builder.Default
    private double paddingRight = 10.0;
    @Min(0)
    @Builder.Default
    private double paddingTop = 20.0;
    @Min(0)
    @Builder.Default
    private double paddingBottom = 20.0;

    @Min(0)
    @Builder.Default
    private double borderLeft = 0.0;
    @Min(0)
    @Builder.Default
    private double borderRight = 0.0;
    @Min(0)
    @Builder.Default
    private double borderTop = 0.0;
    @Min(0)
    @Builder.Default
    private double borderBottom = 0.0;
}

File: frontend/src/app/services/models/column-request.ts
/* tslint:disable */
/* eslint-disable */
import { ColumnSectionRequest } from '../models/column-section-request';
export interface ColumnRequest {
  backgroundColor?: 'PRIMARY' | 'SECONDARY' | 'ACCENT' | 'DARK_BG' | 'LIGHT_BG' | 'DARK_TEXT' | 'LIGHT_TEXT';
  borderBottom?: number;
  borderColor?: 'PRIMARY' | 'SECONDARY' | 'ACCENT' | 'DARK_BG' | 'LIGHT_BG' | 'DARK_TEXT' | 'LIGHT_TEXT';
  borderLeft?: number;
  borderRight?: number;
  borderTop?: number;
  columnNumber?: number;
  id?: number;
  layoutId?: number;
  paddingBottom?: number;
  paddingLeft?: number;
  paddingRight?: number;
  paddingTop?: number;
  sectionMappings: Array<ColumnSectionRequest>;
  textColor?: 'PRIMARY' | 'SECONDARY' | 'ACCENT' | 'DARK_BG' | 'LIGHT_BG' | 'DARK_TEXT' | 'LIGHT_TEXT';
}
